File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, idx, total):
    global answer
    if idx == len(numbers) - 1:
        if total == target:
            answer += 1
        return
    idx += 1
    dfs(numbers, target, idx, total + numbers[idx])
    dfs(numbers, target, idx, total - numbers[idx])
    

def solution(numbers, target):
    global answer
    answer = 0
    dfs(numbers, target, 0, numbers[0])
    dfs(numbers, target, 0, -numbers[0])
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public void dfs(int[] numbers, int target, int idx, int total) {
        if (idx == numbers.length) {
            if (target == total)
                answer++;
            return;
        }
        dfs(numbers, target, idx + 1, total + numbers[idx]);
        dfs(numbers, target, idx + 1, total - numbers[idx]);
    }
    
    
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);   
        
        return answer;
    }
}